### ISSUE TITLE
climate component

### ISSUE THREAD

**Issue by renevdzee** [Bouni/Home-Assistant-Luxtronik#7]
FYI. I will not spend any more time improving my climate component as it will never be possible to set the target temperature from Home-Assistant.

Of course it is not possible to write to a calculated value like ID_Web_RBE_RT_Soll, but I hoped there was some other parameter I could write to set the target temperature. Unfortunately there isn't. 
The Luxtronik module communicates with the RBE through Modbus, it reads the current and set temperatures from the RBE's Modbus input registers (register 0 and 3). Because Modbus input-registers are read-only it will never be possible to modify the set-point.

Same with the 'freigabe' of the cooling (input-register 4), although in that case it might be possible to enable it on the RBE and switch cooling on/off using Luxtronik parameters. 

Feel free to take any code from my repository if you want add a visualization-only climate component to your official Home-Assistant component.

**Comment by Bouni** [comment]
In that case it doen't make sense to add a climate component.

> switch cooling on/off using Luxtronik parameters.
For my Heatpump I just neeed to set [Parameter 108](https://github.com/Bouni/luxtronik/blob/master/luxtronik/parameters.py#L120) to activate cooling.
But that maybe is different for various heatpump types (The reversed code is a fu*** mess of nested if else statements)

Have you ever tried the AlphaInnotec App? I used it also to reverse the functionality and it let me control cooling as well!

Do you have Modbus on your heatpump? I got told that AlphaInnotec charges several 100 bucks for the activation of modbus ...

**Comment by renevdzee** [comment]
Yes, i've tried (and reversed using http://www.javadecompilers.com/apk) the Alpha App, however that app will also not let me change the target room-temperature (just the +/- a few degrees like on the luxtronik screen).

I do not have modbus enabled on my heatpump, but the connection to the RBE uses modbus over RS485. On that bus the Luxtronik is the master and the RBE the slave-device.

**Comment by renevdzee** [comment]
It is a climate component in the real world:
https://www.nathan.nl/regeltechniek/alpha-innotec-ruimtebedieneenheid-rbe/
http://mail.alpha-innotec.de/uploads/Alpha-InnoTec_RBE_finale_Version.pdf

The luxtronik reads the current and target temperature from the RBE, multiplies the difference by a factor (which can be configured) and adjusts the floor temperature by that amount.
It also allows me to view temperatures, enable/disable cooling and set a temporary higher hot water temperature.

**Comment by EveningProgger** [comment]
@renevdzee have to tried to change the needed values in the webinterface (<ip>/Webserver/index.html)? Is it possible there?
I have no RBE

**Comment by renevdzee** [comment]
@EveningProgger The value is readonly and cannot be changed.
The RBE is a rs485 modbus device (kind of) and the value is read from an input register (in theory you could build your own with f.e. an esp8266 and a rs485 transceiver and connect that to homeassistant).

Slightly off-topic, but FYI:
I discovered it is possible to read modbus/tcp input registers 40000 - 40048 from the luxtronik without a license.
I use it with Node-RED (using the node-red-contrib-modbus nodes)
f.e.
- 40039 = TVL
- 40040 = TRL
- 40041 = TA

**Comment by tetsuo55** [comment]
> @EveningProgger The value is readonly and cannot be changed.
> The RBE is a rs485 modbus device (kind of) and the value is read from an input register (in theory you could build your own with f.e. an esp8266 and a rs485 transceiver and connect that to homeassistant).
> 
> Slightly off-topic, but FYI:
> I discovered it is possible to read modbus/tcp input registers 40000 - 40048 from the luxtronik without a license.
> I use it with Node-RED (using the node-red-contrib-modbus nodes)
> f.e.
> 
> * 40039 = TVL
> * 40040 = TRL
> * 40041 = TA

If i understand correctly the rbe moves the whole heatingcurve up or down based on room temperature just like the manual adjustment does and it's also limited to the same values.

We can write to the +/- buttons and use a temperature source from home assistant to move the curve up and down and if that isn't enough we can move the curve itself by changing the top and bottom values in tandem.

Or did someone find a better solution, this beyond my skill to program
